packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { FC } from "react";
+import Modal from "../../../Modal";
+import {
+  DataSourceInterfaceWithParams,
+  ExposureQuery,
+} from "back-end/types/datasource";
+import { useForm } from "react-hook-form";
+import cloneDeep from "lodash/cloneDeep";
+import uniqId from "uniqid";
+import Field from "../../../Forms/Field";
+import CodeTextArea from "../../../Forms/CodeTextArea";
+import Tooltip from "../../../Tooltip";
+import Toggle from "../../../Forms/Toggle";
+import StringArrayField from "../../../Forms/StringArrayField";
+
+type EditExperimentAssignmentQueryProps = {
+  exposureQuery?: ExposureQuery;
+  dataSource: DataSourceInterfaceWithParams;
+  mode: "add" | "edit";
+  onSave: (exposureQuery: ExposureQuery) => void;
+  onCancel: () => void;
+};
+
+export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQueryProps> = ({
+  exposureQuery,
+  dataSource,
+  mode,
+  onSave,
+  onCancel,
+}) => {
+  const modalTitle =
+    mode === "add"
+      ? "Add an Experiment Assignment query"
+      : `Edit ${exposureQuery.name}`;
+
+  const userIdTypeOptions = dataSource.settings.userIdTypes.map(
+    ({ userIdType }) => ({
+      display: userIdType,
+      value: userIdType,
+    })
+  );
+  const defaultUserId = userIdTypeOptions[0]?.value || "user_id";
+
+  const form = useForm<ExposureQuery>({
+    defaultValues:
+      mode === "edit"
+        ? cloneDeep<ExposureQuery>(exposureQuery)
+        : {
+            description: "",
+            id: uniqId("tbl_"),
+            name: "",
+            dimensions: [],
+            query: `SELECT\n  ${defaultUserId} as ${defaultUserId},\n  timestamp as timestamp,\n  experiment_id as experiment_id,\n  variation_id as variation_id\nFROM my_table`,
```

Comment:
`services/datasources.ts` has a function `getInitialSettings` which generates a default query and dimensions array based on the event tracking system they use.  The default query for Segment looks very different than the one for Snowplow for instance.

That `getInitialSettings` function isn't quite right for what you need, but pretty close.

Reply:
I'm pretty sure the `EditDataSourceSettingsForm` is not actually rendered anywhere.  It's referenced from a few old components, but those are all turned off via feature flags.

Reply:
Awesome. I've deleted it here: https://github.com/growthbook/growthbook/pull/550/commits/9fe1c5e331b753fc0ffad9560cfbdba6f7fd0d9d